### ISSUE TITLE
Explain (plan printing) fails for a Hive table partitioned by timestmp or date

### ISSUE THREAD

**Issue by ashwinhs** [prestodb/presto#8329]
Explain <select * from table> gives errors for a table partitoned by timestamp. 

```
CREATE EXTERNAL TABLE TEST_TABLE(
  account_id bigint,
  account_type string,
  account_number bigint
)
PARTITIONED BY (
  business_eff_dt timestamp)
STORED AS ORC
  LOCATION
  's3://fimt-fpphistory-dev-content/TABLE_DATA/TEMP_LOAD/TEST_TABLE'
TBLPROPERTIES (
  'orc.compress'='ZLIB');
```
This is the stack i get when i run explain 
```
presto:default> explain select * from TEST_TABLE;
Query 20170622_135942_00034_dwh5v failed: index (1) must be less than size (1)
java.lang.IndexOutOfBoundsException: index (1) must be less than size (1)
        at com.google.common.base.Preconditions.checkElementIndex(Preconditions.java:1177)
        at com.google.common.base.Preconditions.checkElementIndex(Preconditions.java:1159)
        at com.google.common.collect.SingletonImmutableList.get(SingletonImmutableList.java:41)
        at com.facebook.presto.sql.FunctionInvoker.invoke(FunctionInvoker.java:67)
        at com.facebook.presto.sql.FunctionInvoker.invoke(FunctionInvoker.java:43)
        at com.facebook.presto.sql.planner.PlanPrinter.castToVarchar(PlanPrinter.java:1376)
        at com.facebook.presto.sql.planner.PlanPrinter.access$700(PlanPrinter.java:146)
        at com.facebook.presto.sql.planner.PlanPrinter$Visitor.lambda$formatDomain$15(PlanPrinter.java:1293)
        at com.facebook.presto.spi.predicate.SortedRangeSet$2.consume(SortedRangeSet.java:220)
        at com.facebook.presto.sql.planner.PlanPrinter$Visitor.formatDomain(PlanPrinter.java:1288)
        at com.facebook.presto.sql.planner.PlanPrinter$Visitor.printConstraint(PlanPrinter.java:1274)
        at com.facebook.presto.sql.planner.PlanPrinter$Visitor.printTableScanInfo(PlanPrinter.java:1036)
        at com.facebook.presto.sql.planner.PlanPrinter$Visitor.visitTableScan(PlanPrinter.java:909)
        at com.facebook.presto.sql.planner.PlanPrinter$Visitor.visitTableScan(PlanPrinter.java:620)
        at com.facebook.presto.sql.planner.plan.TableScanNode.accept(TableScanNode.java:135)
        at com.facebook.presto.sql.planner.PlanPrinter$Visitor.processChildren(PlanPrinter.java:1247)
        at com.facebook.presto.sql.planner.PlanPrinter$Visitor.visitExchange(PlanPrinter.java:1189)
        at com.facebook.presto.sql.planner.PlanPrinter$Visitor.visitExchange(PlanPrinter.java:620)
        at com.facebook.presto.sql.planner.plan.ExchangeNode.accept(ExchangeNode.java:195)
        at com.facebook.presto.sql.planner.PlanPrinter$Visitor.processChildren(PlanPrinter.java:1247)
        at com.facebook.presto.sql.planner.PlanPrinter$Visitor.visitOutput(PlanPrinter.java:1077)
        at com.facebook.presto.sql.planner.PlanPrinter$Visitor.visitOutput(PlanPrinter.java:620)
        at com.facebook.presto.sql.planner.plan.OutputNode.accept(OutputNode.java:82)
        at com.facebook.presto.sql.planner.PlanPrinter.<init>(PlanPrinter.java:167)
        at com.facebook.presto.sql.planner.PlanPrinter.<init>(PlanPrinter.java:154)
        at com.facebook.presto.sql.planner.PlanPrinter.textLogicalPlan(PlanPrinter.java:191)
        at com.facebook.presto.sql.analyzer.QueryExplainer.getPlan(QueryExplainer.java:95)
        at com.facebook.presto.sql.rewrite.ExplainRewrite$Visitor.getQueryPlan(ExplainRewrite.java:126)
        at com.facebook.presto.sql.rewrite.ExplainRewrite$Visitor.visitExplain(ExplainRewrite.java:104)
        at com.facebook.presto.sql.rewrite.ExplainRewrite$Visitor.visitExplain(ExplainRewrite.java:60)
        at com.facebook.presto.sql.tree.Explain.accept(Explain.java:80)
        at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:27)
        at com.facebook.presto.sql.rewrite.ExplainRewrite.rewrite(ExplainRewrite.java:57)
        at com.facebook.presto.sql.rewrite.StatementRewrite.rewrite(StatementRewrite.java:51)
        at com.facebook.presto.sql.analyzer.Analyzer.analyze(Analyzer.java:65)
        at com.facebook.presto.sql.analyzer.Analyzer.analyze(Analyzer.java:60)
        at com.facebook.presto.execution.SqlQueryExecution.doAnalyzeQuery(SqlQueryExecution.java:296)
        at com.facebook.presto.execution.SqlQueryExecution.analyzeQuery(SqlQueryExecution.java:282)
        at com.facebook.presto.execution.SqlQueryExecution.start(SqlQueryExecution.java:238)
        at com.facebook.presto.execution.QueuedExecution.lambda$start$1(QueuedExecution.java:62)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:748)
```

This looks like its happening for a "timestamp" or "date" partitioned table. I tried on a table without partitioning table and another table with "string" partitioning and both worked fine without issues. Note: The tables are Hive tables, stored as ORC accessed in presto via the hive connector.

**Comment by maciejgrzybek** [comment]
Patch is ready here: https://github.com/prestodb/presto/pull/8502

**Comment by ashwinhs** [comment]
Was this fixed in 0.181 ?

**Comment by maciejgrzybek** [comment]
No, there's no released version of Presto having this patch yet.
0.182 [unreleased] and 179t of Teradata's distro [soon to be released] will be first versions with this patch.
